coordinator/src/main.rs

```diff
@@ -0,0 +1,300 @@
+                }
+            }
+            paladin_runtime => {
+                info!(
+                    "Not initializing prover_state_manager due to Paladin Runtime: {:?}",
+                    paladin_runtime
+                );
+            }
+        }
+        info!("Building Paladin Runtime");
+        match Runtime::from_config(&config, register()).await {
+            Ok(runtime) => {
+                info!("Created Paladin Runtime");
+                runtime
+            }
+            Err(err) => {
+                error!("Config: {:#?}", config);
+                error!("Error while constructing the runtime: {}", err);
+                panic!("Failed to build Paladin runtime from config: {}", err);
+            }
+        }
+    };
+
+    // Store it in an Arc
+    let webdata_runtime = web::Data::new(runtime);
+
+    //------------------------------------------------------------------------
+    // Server
+    //------------------------------------------------------------------------
+
+    debug!("Setting up server endpoint");
+
+    let server_addr = match env::var(SERVER_ADDR_ENVKEY) {
+        Ok(addr) => {
+            info!("Retrieved server address: {}", addr);
+            addr
+        }
+        Err(env::VarError::NotPresent) => {
+            warn!("Using default server address: {}", DFLT_SERVER_ADDR);
+            String::from(DFLT_SERVER_ADDR)
+        }
+        Err(env::VarError::NotUnicode(os_str)) => {
+            error!("Non-unicode server address: {:?}", os_str);
+            panic!("Non-unicode server address: {:?}", os_str);
+        }
+    };
+
+    info!("Starting HTTP Server: {}", server_addr);
+
+    // Set up the server
+    let server = match HttpServer::new(move || {
+        App::new()
+            .app_data(webdata_runtime.clone())
+            .service(web::resource("/").route(web::post().to(handle_post)))
+            .route("/health", web::get().to(handle_health))
+    })
+    .bind(server_addr)
+    {
+        Ok(item) => item,
+        Err(err) => panic!("Failed to start the server: {}", err),
+    };
+
+    // Run the server, panic if there's an error with it.
+    if let Err(err) = server.run().await {
+        error!("Error with running the server: {}", err);
+        panic!("Failed to run the server: {}", err);
+    }
+
+    Ok(())
+}
+
+/// Returns [HttpResponse] ([HttpResponse::Ok]) to respond that we are healthy
+async fn handle_health() -> impl Responder {
+    info!("Received health check, responding `OK`");
+    HttpResponse::Ok().body("OK")
+}
+
+/// Recevies a request for [manyprover::prove_blocks]
+async fn handle_post(
+    runtime: web::Data<Runtime>,
+    input: web::Json<input::ProveBlocksInput>,
+) -> impl Responder {
+    let start_time = match SystemTime::now().duration_since(UNIX_EPOCH) {
+        Ok(duration) => duration.as_secs(),
+        Err(err) => {
+            warn!(
+                "Unable to determine current time, using 0 for current seconds: {}",
```

Comment:
I think that warning and moving on with a value of 0 is dangerous. 

Getting the system time **should not fail** under any circumstances, so if it does, there's probably something really really wrong. I would explicitly `panic()` in this case to make sure that you don't miss the error message (rather than wondering why a result somewhere is weird and having to dig through all of the logs to investigate).